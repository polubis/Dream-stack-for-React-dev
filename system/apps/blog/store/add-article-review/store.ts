import { create } from 'zustand';
import type { AddArticleReviewStore } from './defs';
import { addArticleReviewForm } from './core';

const useAddArticleReviewStore = create<AddArticleReviewStore.State>(() => ({
  is: 'idle',
  form: addArticleReviewForm.init({ content: '' }),
}));

export { useAddArticleReviewStore, addArticleReviewForm };
