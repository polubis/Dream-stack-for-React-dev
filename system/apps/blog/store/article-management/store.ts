import { create } from 'zustand';
import type * as ArticleManagement from './defs';

const useArticleManagementStore = create<ArticleManagement.State>(() => ({
  is: 'idle',
}));

export { useArticleManagementStore };
