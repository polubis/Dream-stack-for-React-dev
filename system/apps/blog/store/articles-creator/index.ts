export * from './articles-creator.store';
export * from './defs';
