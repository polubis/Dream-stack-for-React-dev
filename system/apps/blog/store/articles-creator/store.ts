import { create } from 'zustand';
import type { ArticlesCreator } from './defs';
import { creatorForm } from './form';

const useArticlesCreatorStore = create<ArticlesCreator.State>(() => ({
  is: 'idle',
  view: 'initial',
  form: creatorForm.init({
    title: '',
    description: '',
    tags: [],
    thumbnail: {
      file: null,
      preview: [],
    },
    content: '',
    lang: 'en',
    sendToReview: false,
  }),
}));

export { useArticlesCreatorStore };
