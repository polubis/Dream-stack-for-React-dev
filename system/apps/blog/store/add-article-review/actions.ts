import { createArticleReview, getError } from '@system/blog-api';
import type * as AddArticleReview from './defs';
import { addArticleReviewForm, useAddArticleReviewStore } from './store';

const { setState: set, getState: get } = useAddArticleReviewStore;

const add_article_review_actions: AddArticleReview.Actions = {
  setField: (key, value) => {
    set({ form: addArticleReviewForm.set(get().form)({ [key]: value }) });
  },
  confirm: async (id) => {
    const form = addArticleReviewForm.confirm(get().form);

    if (form.invalid) {
      set({ form });
      return;
    }

    set({ is: 'busy' });

    try {
      const response = await createArticleReview({
        id,
        ...get().form.values,
      });

      set({ is: 'ok' });
      add_article_review_actions.setField('content', '');

      return response;
    } catch (error: unknown) {
      set({ is: 'fail', error: getError(error) });
    }
  },
};

export { add_article_review_actions };
