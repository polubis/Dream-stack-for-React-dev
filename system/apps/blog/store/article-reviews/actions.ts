import { getArticleReviews, getError } from '@system/blog-api';
import type * as ArticleReviews from './defs';
import { useArticleReviewsStore } from './store';

const { setState: set, getState: get } = useArticleReviewsStore;

const article_reviews_actions: ArticleReviews.Actions = {
  load: async (id) => {
    set({ is: 'busy' }, true);

    try {
      const { data } = await getArticleReviews({ id });
      set({ is: 'ok', reviews: data }, true);
    } catch (error: unknown) {
      set({ is: 'fail', error: getError(error) }, true);
    }
  },
  addReview: (review) => {
    const state = get();

    if (state.is === 'ok') {
      set({ reviews: [...state.reviews, review] });
      return;
    }

    throw Error('You are trying to add review when there is no data yet');
  },
  reset: () => {
    set({ is: 'idle' }, true);
  },
};

export { article_reviews_actions };
