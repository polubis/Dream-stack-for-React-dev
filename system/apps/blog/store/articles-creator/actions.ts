import { useArticlesCreatorStore } from './store';
import type { ArticlesCreator } from './defs';
import { creatorForm } from './form';
import {
  createArticle,
  getError,
  sendArticleForApproval,
  updateArticle,
} from '@system/blog-api';
import { Url } from '@system/blog-api-models';

const { getState: get, setState: set } = useArticlesCreatorStore;

const articles_creator_actions = {
  setView: (view: ArticlesCreator.View): void => {
    set({ view });
  },
  setForm: (values: Partial<ArticlesCreator.FormData> = {}): void => {
    set({
      form: creatorForm.init({
        ...get().form.values,
        ...values,
      }),
    });
  },
  change: <
    K extends keyof ArticlesCreator.FormData,
    V extends ArticlesCreator.FormData[K]
  >(
    key: K,
    value: V
  ): void => {
    set({
      form: creatorForm.set(get().form)({ [key]: value }),
    });
  },
  confirm: async (url?: Url): Promise<void> => {
    const state = get();

    set({
      is: 'busy',
      form: creatorForm.confirm(state.form),
    });

    try {
      const {
        form: {
          values: { title, thumbnail, description, content, lang },
        },
      } = state;

      const payload = {
        title,
        thumbnail: thumbnail.file,
        description,
        content,
        lang,
      };

      const { data } = await (url
        ? updateArticle({ url, ...payload })
        : createArticle(payload));

      if (state.form.values.sendToReview) {
        await sendArticleForApproval({ id: data.id });
      }

      set({ is: 'ok' });
    } catch (error: unknown) {
      set({ is: 'fail', error: getError(error) });
    }
  },
};

export { articles_creator_actions };
