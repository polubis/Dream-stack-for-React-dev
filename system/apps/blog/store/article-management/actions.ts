import { useArticleManagementStore } from './store';
import type * as ArticleManagement from './defs';
import { article_reviews_actions } from '../article-reviews';
import { article_store_actions } from '../article/actions';
import { add_article_review_store_actions } from '../add-article-review';
import { mockArticleReview } from '@system/blog-api-mocks';
import { change_article_status_actions } from '../change-article-status';

const { setState: set } = useArticleManagementStore;

const article_management_actions: ArticleManagement.Actions = {
  start: (id, url, lang) => {
    set({ is: 'active', id });
    article_reviews_actions.load(id);
    article_store_actions.load({ url, lang });
  },
  sendForApproval: async (id) => {
    await change_article_status_actions.sendForApproval(id);
    article_store_actions.update({ status: 'WaitingForApproval' });
    return;
  },
  reset: () => {
    set({ is: 'idle' });
  },
  confirm: async (id) => {
    await add_article_review_store_actions.confirm(id);
    // @TODO: Remove mock when backend will finish work.
    article_reviews_actions.addReview(mockArticleReview());
  },
  changeStatus: async (id, status) => {
    if (status === 'Accepted') {
      await change_article_status_actions.accept(id);
      article_store_actions.update({ status });
      return;
    }

    if (status === 'NeedWork') {
      await change_article_status_actions.reject(id);
      article_store_actions.update({ status });
      return;
    }
  },
};

export { article_management_actions };
