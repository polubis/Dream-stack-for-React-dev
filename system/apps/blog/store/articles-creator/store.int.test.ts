import { getPath, requestFixture } from '@system/blog-api';
import { storeFixture } from '../test-utils';
import { useArticlesCreatorStore } from './store';
import { act, waitFor } from '@testing-library/react';
import { articles_creator_actions } from './actions';
import {
  mockErrorResponse,
  mockGetArticleResponse,
} from '@system/blog-api-mocks';
import type { ArticlesCreator } from './defs';
import type { Lang, Url } from '@system/blog-api-models';

describe('Articles creation feature works when: ', () => {
  const url: Url = 'some-url';
  const lang: Lang = 'en';

  const { clean, server, rest } = requestFixture();

  beforeAll(() => {
    server.listen();
  });

  afterEach(() => {
    clean();
  });

  afterAll(() => {
    server.close();
  });

  it('allows to update loaded article', async () => {
    const article = mockGetArticleResponse();
    server.use(
      rest.get([getPath('Articles'), lang, url].join('/'), (_, res, ctx) => {
        return res(ctx.status(200), ctx.json(article));
      })
    );
    server.use(
      rest.put([getPath('Articles'), lang, url].join('/'), (_, res, ctx) => {
        return res(ctx.status(200));
      })
    );

    const { result, restore } = storeFixture(useArticlesCreatorStore);

    act(() => {
      articles_creator_actions.init(lang, url);
    });

    await waitFor(() => {
      expect(result.current.is).toBe('edition');
      expect(result.current).toMatchSnapshot();
    });

    act(() => {
      articles_creator_actions.change('thumbnail', {
        file: {} as File,
        preview: [article.data.thumbnailUrl],
      });
    });

    await waitFor(() => {
      expect(result.current.is).toBe('edition');
      expect(result.current).toMatchSnapshot();
    });

    act(() => {
      articles_creator_actions.confirm();
    });

    expect(result.current.is).toBe('editing');

    await waitFor(() => {
      expect(result.current.is).toBe('edited');
      expect(result.current).toMatchSnapshot();
    });

    restore();
  });

  it('handles error if article update fail', async () => {
    const article = mockGetArticleResponse();
    server.use(
      rest.get([getPath('Articles'), lang, url].join('/'), (_, res, ctx) => {
        return res(ctx.status(200), ctx.json(article));
      })
    );
    server.use(
      rest.put([getPath('Articles'), lang, url].join('/'), (_, res, ctx) => {
        return res(ctx.status(400), ctx.json(mockErrorResponse()));
      })
    );

    const { result, restore } = storeFixture(useArticlesCreatorStore);

    act(() => {
      articles_creator_actions.init(lang, url);
    });

    await waitFor(() => {
      expect(result.current.is).toBe('edition');
      expect(result.current).toMatchSnapshot();
    });

    act(() => {
      articles_creator_actions.change('thumbnail', {
        file: {} as File,
        preview: [article.data.thumbnailUrl],
      });
    });

    await waitFor(() => {
      expect(result.current.is).toBe('edition');
      expect(result.current).toMatchSnapshot();
    });

    act(() => {
      articles_creator_actions.confirm();
    });

    expect(result.current.is).toBe('editing');

    await waitFor(() => {
      expect(result.current.is).toBe('edit-fail');
      expect(result.current).toMatchSnapshot();
    });

    restore();
  });

  it('handles load edition data error', async () => {
    server.use(
      rest.get([getPath('Articles'), lang, url].join('/'), (_, res, ctx) => {
        return res(ctx.status(400), ctx.json(mockErrorResponse()));
      })
    );

    const { result, restore } = storeFixture(useArticlesCreatorStore);

    act(() => {
      articles_creator_actions.init(lang, url);
    });

    expect(result.current.is).toBe('loading');
    expect((result.current as ArticlesCreator.Loading).url).toBe(url);

    await waitFor(() => {
      expect(result.current.is).toBe('load-fail');
      expect(result.current).toMatchSnapshot();
    });

    restore();
  });

  it('loads edition data when url passed', async () => {
    server.use(
      rest.get([getPath('Articles'), lang, url].join('/'), (_, res, ctx) => {
        return res(ctx.status(200), ctx.json(mockGetArticleResponse()));
      })
    );

    const { result, restore } = storeFixture(useArticlesCreatorStore);

    act(() => {
      articles_creator_actions.init(lang, url);
    });

    expect(result.current.is).toBe('loading');
    expect((result.current as ArticlesCreator.Loading).url).toBe(url);

    await waitFor(() => {
      expect(result.current.is).toBe('edition');
      expect(result.current).toMatchSnapshot();
    });

    restore();
  });

  it('handles creation error', async () => {
    server.use(
      rest.post(getPath('Articles'), (_, res, ctx) => {
        return res(ctx.status(400), ctx.json(mockErrorResponse()));
      })
    );

    const { result, restore } = storeFixture(useArticlesCreatorStore);

    act(() => {
      articles_creator_actions.init(lang);
      articles_creator_actions.change('title', 'my-title');
      articles_creator_actions.change('thumbnail', {
        file: {} as File,
        preview: [],
      });
      articles_creator_actions.confirm();
    });

    expect(result.current.is).toBe('creating');

    await waitFor(() => {
      expect(result.current.is).toBe('create-fail');
      expect(result.current).toMatchSnapshot();
    });

    restore();
  });

  it('creates article', async () => {
    server.use(
      rest.post(getPath('Articles'), (_, res, ctx) => {
        return res(ctx.status(200));
      })
    );

    const { result, restore } = storeFixture(useArticlesCreatorStore);

    act(() => {
      articles_creator_actions.init(lang);
      articles_creator_actions.change('title', 'my-title');
      articles_creator_actions.change('thumbnail', {
        file: {} as File,
        preview: [mockGetArticleResponse().data.thumbnailUrl],
      });
      articles_creator_actions.confirm();
    });

    expect(result.current.is).toBe('creating');

    await waitFor(() => {
      expect(result.current.is).toBe('created');
      expect(result.current).toMatchSnapshot();
    });

    restore();
  });
});
