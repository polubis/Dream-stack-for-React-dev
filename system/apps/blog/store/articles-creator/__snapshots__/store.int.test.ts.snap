// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles creation feature works when:  allows to update loaded article 1`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 0,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touched": false,
    "untouched": true,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": null,
        "preview": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edition",
  "url": "some-url",
}
`;

exports[`Articles creation feature works when:  allows to update loaded article 2`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 100,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": true,
      "title": false,
    },
    "touched": true,
    "untouched": false,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": Object {},
        "preview": Array [
          "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
        ],
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edition",
  "url": "some-url",
}
`;

exports[`Articles creation feature works when:  allows to update loaded article 3`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 100,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": true,
      "title": false,
    },
    "touched": true,
    "untouched": false,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": Object {},
        "preview": Array [
          "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
        ],
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edited",
  "url": "some-url",
}
`;

exports[`Articles creation feature works when:  creates article 1`] = `
Object {
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 100,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": true,
      "title": true,
    },
    "touched": true,
    "untouched": false,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "",
      "description": "",
      "lang": "en",
      "thumbnail": Object {
        "file": Object {},
        "preview": Array [
          "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
        ],
      },
      "title": "my-title",
    },
  },
  "is": "created",
}
`;

exports[`Articles creation feature works when:  handles creation error 1`] = `
Object {
  "error": Object {
    "key": "unknown",
    "message": "Something went wrong...",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 100,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": true,
      "title": true,
    },
    "touched": true,
    "untouched": false,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "",
      "description": "",
      "lang": "en",
      "thumbnail": Object {
        "file": Object {},
        "preview": Array [],
      },
      "title": "my-title",
    },
  },
  "is": "create-fail",
}
`;

exports[`Articles creation feature works when:  handles error if article update fail 1`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 0,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touched": false,
    "untouched": true,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": null,
        "preview": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edition",
  "url": "some-url",
}
`;

exports[`Articles creation feature works when:  handles error if article update fail 2`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 100,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": true,
      "title": false,
    },
    "touched": true,
    "untouched": false,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": Object {},
        "preview": Array [
          "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
        ],
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edition",
  "url": "some-url",
}
`;

exports[`Articles creation feature works when:  handles error if article update fail 3`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "error": Object {
    "key": "unknown",
    "message": "Something went wrong...",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 100,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": true,
      "title": false,
    },
    "touched": true,
    "untouched": false,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": Object {},
        "preview": Array [
          "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
        ],
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edit-fail",
  "url": "some-url",
}
`;

exports[`Articles creation feature works when:  handles load edition data error 1`] = `
Object {
  "error": Object {
    "key": "unknown",
    "message": "Something went wrong...",
  },
  "is": "load-fail",
}
`;

exports[`Articles creation feature works when:  loads edition data when url passed 1`] = `
Object {
  "article": Object {
    "authorEmail": "Polubis@greenonsoftware.com",
    "authorName": "Polubis",
    "content": "# Markdown content",
    "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
    "id": "c54cefcb-ccd0-4b94-f68f-08db5ed3b360",
    "lang": "en",
    "status": "Accepted",
    "thumbnailUrl": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
    "title": "Porting for React applications",
    "url": "porting-for-react-applications",
  },
  "form": Object {
    "dirty": false,
    "errors": Object {
      "content": null,
      "description": null,
      "lang": null,
      "thumbnail": null,
      "title": null,
    },
    "invalid": false,
    "invalidCount": 0,
    "keys": Array [
      "title",
      "description",
      "thumbnail",
      "content",
      "lang",
    ],
    "pristine": true,
    "progress": 0,
    "result": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touch": Object {
      "content": false,
      "description": false,
      "lang": false,
      "thumbnail": false,
      "title": false,
    },
    "touched": false,
    "untouched": true,
    "valid": true,
    "validCount": 5,
    "values": Object {
      "content": "# Markdown content",
      "description": "Let's see how you can save time by using the porting technique when migrating from one technology to another.",
      "lang": "en",
      "thumbnail": Object {
        "file": null,
        "preview": "https://greenonsoftwarestorage.blob.core.windows.net/dev/tmpE09_202307171409592.png",
      },
      "title": "Porting for React applications",
    },
  },
  "is": "edition",
  "url": "some-url",
}
`;
